Add note about standalone + runtimeConfig caveat (#38532)
vercel/next.js: docs/advanced-features/output-file-tracing.md

Additionally, a minimal `server.js` file is also output which can be used instead of `next start`. This minimal server does not copy the `public` or `.next/static` folders by default as these should ideally be handled by a CDN instead, although these folders can be copied to the `standalone/public` and `standalone/.next/static` folders manually, after which `server.js` file will serve these automatically.

> Note: `next.config.js` is serialized when creating the minimal `server.js` file so if the legacy [`serverRuntimeConfig` or `publicRuntimeConfig` options](/docs/api-reference/next.config.js/runtime-configuration.md) are being used the values will be specific to the build time provided values.

Reviewer: ```suggestion
Note: `next.config.js` is read during `next build` and serialized in the `server.js` output file. If the legacy [`serverRuntimeConfig` or `publicRuntimeConfig` options](/docs/api-reference/next.config.js/runtime-configuration.md) are being used, the values will be specific to values at build time.
```
